lua/entities/info_wiremapinterface/entityoverride.lua

```diff
@@ -1,71 +1,868 @@
+	end
+
+	self:_WMI_ApplyWireDupeData(ply, wireDupeInfo, createdEntities, tmpPorts)
+	interfaceEnt:AddSingleEntity(self)
+end
+
+local function GetEntityLookupFunction(createdEntities)
+	return function(idx, default)
+		if idx == nil then
+			return default
+		end
+
+		if idx == 0 then
+			return game.GetWorld()
+		end
+
+		local ent = createdEntities[idx]
+		if IsValid(ent) then
+			return ent
+		end
+
+		return default
+	end
+end
+
+function WIREENT:_WMI_ApplyWireDupeData(ply, wireDupeInfo, createdEntities, tmpPorts)
+	self:_WMI_AddTmpPorts(tmpPorts)
+
+	local lookupFunc = GetEntityLookupFunction(createdEntities)
+	WireLib.ApplyDupeInfo(ply, self, wireDupeInfo, lookupFunc)
+end
+
+function WIREENT:_WMI_MapCreationIdDuped()
+	local wmidata = self._WireMapInterfaceEnt_Data
+
+	if wmidata then
+		local dupe = wmidata.dupe
+
+		if dupe and dupe.mapId then
+			return dupe.mapId
+		end
+	end
+
+	local mapId = self._WireMapInterfaceEnt_MapId
+	if mapId then
+		return mapId
+	end
+
+	return self:MapCreationID()
+end
 
-	local Input = self.Inputs[name] or {}
-	Entity:TriggerWireInput(name, value, IsValid(Input.Src) == true, self, ...)
+function WIREENT:_WMI_GetSpawnIdDuped(interfaceEnt)
+	local wmidata = self._WireMapInterfaceEnt_Data
+
+	if wmidata then
+		local dupe = wmidata.dupe
+
+		if dupe and dupe.spawnId then
+			return dupe.spawnId
+		end
+	end
+
+	return WIREENT._WMI_GetSpawnId(self, interfaceEnt)
+end
+
+function WIREENT:_WMI_GetSpawnId(interfaceEnt)
+	if self._WireMapInterfaceEnt_SpawnId then
+		return self._WireMapInterfaceEnt_SpawnId
+	end
+
+	self._WireMapInterfaceEnt_SpawnId = nil
+
+	if not IsValid(interfaceEnt) then
+		local interfaceEnt = self._WireMapInterfaceEnt
+
+		if not IsValid(interfaceEnt) then
+			local wmidata = self._WireMapInterfaceEnt_Data
+			if not wmidata then
+				return nil
+			end
+
+			local dupe = wmidata.dupe
+			if not dupe then
+				return nil
+			end
+
+			interfaceEnt = dupe.interfaceEnt
+			if not IsValid(interfaceEnt) then
+				return nil
+			end
+		end
+	end
+
+	local id = interfaceEnt:HashSubEntityMapId(
+		self:EntIndex(),
+		WIREENT._WMI_MapCreationIdDuped(self)
+	)
+
+	self._WireMapInterfaceEnt_SpawnId = id
+
+	return id
 end
 
 -- Remove its overrides
-function WIREENT:_RemoveOverrides()
-	for k, v in pairs(self._Overrides_WireMapInterfaceEnt or {}) do
-		self[k] = v
+function WIREENT:_WMI_RemoveOverrides(interfaceEnt)
+	local wmidata = self._WireMapInterfaceEnt_Data
+
+	if not IsValid(interfaceEnt) then
+		interfaceEnt = self._WireMapInterfaceEnt
+	end
+
+	self:RemoveCallOnRemove("WireMapInterface_OnRemove")
+	duplicator.ClearEntityModifier(self, "WireDupeInfo")
+	duplicator.ClearEntityModifier(self, "WireMapInterfaceEntDupeInfo")
+
+	local spawnId = WIREENT._WMI_GetSpawnId(self, interfaceEnt)
+	local spawnIdDuped = WIREENT._WMI_GetSpawnIdDuped(self, interfaceEnt)
+	local mapId = WIREENT._WMI_MapCreationIdDuped(self)
+
+	WireLib.UnregisterWireMapInterfaceSpawnId(spawnId, spawnIdDuped, mapId, self)
```

Comment:
That's where it is called to unregister itself from the lookup table. `WIREENT:_WMI_RemoveOverrides` is called if the entity is removed or is being unassigned from it's `info_wiremapinterface`.